File: sdk/php/src/Resolve.php
<?php

declare(strict_types=1);

namespace Noizu\DirenvConfig;

use Symfony\Component\Yaml\Yaml;

final class Resolve
{
    /**
     * Merge layer files and write the resolved .active file.
     *
     * Layer order: base.yaml → {DC_ENV}.yaml → local.yaml → secrets.yaml
     */
    public static function resolveActive(string $storePath, string $name): mixed
    {
        $configDir = $storePath . '/' . $name;
        $env = getenv('DC_ENV') ?: 'dev';

        $layerFiles = [
            $configDir . '/base.yaml',
            $configDir . '/' . $env . '.yaml',
            $configDir . '/local.yaml',
            $configDir . '/secrets.yaml',
        ];

        $layers = [];
        foreach ($layerFiles as $file) {
            if (file_exists($file)) {
                $content = file_get_contents($file);
                $parsed = Yaml::parse($content);
                if ($parsed !== null) {
                    $layers[] = $parsed;
                }
            }
        }

        $merged = count($layers) > 0 ? Merge::deepMergeMulti($layers) : [];

        // Ensure config dir exists
        if (!is_dir($configDir)) {
            mkdir($configDir, 0755, true);
        }

        $activePath = $configDir . '/.active';
        file_put_contents($activePath, Yaml::dump($merged, 4, 2));

        return $merged;
    }
}

File: sdk/python/src/direnv_config/resolve.py
from __future__ import annotations

import os
from pathlib import Path
from typing import Any

import yaml

from direnv_config.merge import deep_merge_multi


def resolve_active(store_path: Path, name: str) -> Any:
    """Merge config layers and write the result to .active.

    Layer merge order: base.yaml -> {DC_ENV}.yaml -> local.yaml -> secrets.yaml
    """
    config_dir = store_path / name
    config_dir.mkdir(parents=True, exist_ok=True)

    dc_env = os.environ.get("DC_ENV", "dev")

    layer_names = ["base", dc_env, "local", "secrets"]
    layers: list[Any] = []

    for layer_name in layer_names:
        layer_file = config_dir / f"{layer_name}.yaml"
        if layer_file.exists():
            contents = layer_file.read_text()
            data = yaml.safe_load(contents)
            if data is not None:
                layers.append(data)

    merged = deep_merge_multi(layers)

    active_file = config_dir / ".active"
    if merged is not None:
        active_file.write_text(yaml.dump(merged, default_flow_style=False))
    else:
        active_file.write_text("")

    return merged
